atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java

```diff
@@ -134,80 +134,107 @@ static void validatePartitioner(Cassandra.Client client, CassandraKeyValueServic
 
     static void ensureKeyspaceExistsAndIsUpToDate(CassandraClientPool clientPool, CassandraKeyValueServiceConfig config)
             throws TException {
+        createKeyspace(config);
+        updateExistingKeyspace(clientPool, config);
+    }
+
+    // swallows the expected TException subtype NotFoundException, throws connection problem related ones
+    private static boolean keyspaceAlreadyExists(InetSocketAddress host, CassandraKeyValueServiceConfig config)
+            throws TException {
         try {
-            clientPool.run(new FunctionCheckedException<Cassandra.Client, Void, TException>() {
-                @Override
-                public Void apply(Cassandra.Client client) throws TException {
-                    KsDef originalKsDef = client.describe_keyspace(config.keyspace());
-                    KsDef modifiedKsDef = originalKsDef.deepCopy();
-                    CassandraVerifier.checkAndSetReplicationFactor(
-                            client,
-                            modifiedKsDef,
-                            false,
-                            config.replicationFactor(),
-                            config.safetyDisabled());
-
-                    if (!modifiedKsDef.equals(originalKsDef)) {
-                        // Can't call system_update_keyspace to update replication factor if CfDefs are set
-                        modifiedKsDef.setCf_defs(ImmutableList.of());
-                        client.system_update_keyspace(modifiedKsDef);
-                        CassandraKeyValueServices.waitForSchemaVersions(
-                                client,
-                                "(updating the existing keyspace)",
-                                config.schemaMutationTimeoutMillis());
-                    }
-
-                    return null;
+            Client client = CassandraClientFactory.getClientInternal(host, config);
+            client.describe_keyspace(config.keyspace());
+        } catch (NotFoundException e) {
+            return false;
+        }
+        return true;
+    }
+
+    private static void createKeyspace(CassandraKeyValueServiceConfig config) throws TException {
+        // We can't use the pool yet because it does things like setting the connection's keyspace for us
+        boolean someHostWasAbleToCreateTheKeyspace = false;
```

Comment:
Not entirely within the scope of this PR, but if we move the for loop into a separate method, then it might look a bit cleaner. I'll try out a quick refactor once this merges.